app/codecept/scenarios/gds_moj_components_render_test.js

```diff
@@ -0,0 +1,18 @@
+Feature('GDS and MOJ components are available');
+
+Scenario('displays a GDS summary element', ({ I}) => {
+    I.amOnPage('/home')
+    I.seeElement(
+        locate('summary').withText('Help with nationality')
```

Comment:
Locating via text is a pretty brittle pattern in e2e tests. Might be worth getting test hooks via data attributes and a constants file in place as part of this work to avoid issues later.

Reply:
In this example, I'd agree but I think there are some situations where locating by text is actually helpful as its giving an assertion for free on if something has gone wrong with rendering text on a component that has a design that is expected to be static.

As much as I don't like adding things to the DOM just for testing purposes I can see the value in some high-level data-* attributes we can hook into where we need to. Do you have some examples of how other teams have approached this?

Reply:
It comes down a bit to "what are you testing with this?" Sometimes you might want to check the text is "right", so on translations and errors, but other times it's outside the scope. Also means message catalogue changes are coupled to deploys, which you might not want if you have a bigger content/translation team working outside of the dev process.

I put some examples of data attribute use in the testing with cypress doc. They don't need to be test specific, in fact I avoid `data-qa-hook` personally, I tend toward the pattern of additional app semantics, like `data-action="client.1.edit"` which convey intent of the page element.

Reply:
I'll leave this test as is as it's here to prove everything is hooked up right and will be dropped when we write our first real test but will apply the `data-*` approach for the login flow.

Reply:
I'll leave this test as is as it's here to prove everything is hooked up right and will be dropped when we write our first real test but will apply the data-* approach for the login flow.